src/bracketing.jl

```diff
@@ -826,6 +826,118 @@ function update_state(
     return state, false
 end
 
+## --------------------------------------------------
+
+"""
+    Roots.ITP(;[κ₁, κ₂, n₀])
+
+Use the [ITP](https://en.wikipedia.org/wiki/ITP_method) bracketing
+method.  This method claims it "is the first root-finding algorithm
+that achieves the superlinear convergence of the secant method[1]
+while retaining the optimal[2] worst-case performance of the bisection
+method."
+
+The values `κ1`, `κ₂`, and `n₀` are tuning parameters.
+
+The [suggested](https://docs.rs/kurbo/0.8.1/kurbo/common/fn.solve_itp.html) value of `κ₁` is `0.2/(b-a)`, but the default here is `0.2`. The default value of `κ₂` is `2`, and the default value of `n₀=1`.
+
+## Note:
+
+Suggested on [discourse](https://discourse.julialang.org/t/julia-implementation-of-the-interpolate-truncate-project-itp-root-finding-algorithm/77739) by `@TheLateKronos`, who supplied the original version of the code below.
+"""
+struct ITP{T,S} <: AbstractAlefeldPotraShi
+    κ₁::T
+    κ₂::S
+    n₀::Int
+    function ITP(κ₁::T′, κ₂::S, n₀::Int) where {T′, S}
+        0 ≤ κ₁ < Inf             ||   throw(ArgumentError("κ₁ must be between 0 and ∞"))
+        1 ≤ κ₂ < 1+ (1 + √5)/2   ||   throw(ArgumentError("κ₂ must be between 1 and the golden ratio"))
```

Comment:
```suggestion
        1 ≤ κ₂ < 1+ (1 + √5)/2   ||   throw(ArgumentError("κ₂ must be between 1 and 1+ (1 + √5)/2 (1 + the golden ratio)"))
```

Reply:
I happened to make a mistake in the original error message. This fixes it ^_^

Reply:
note that `(3+sqrt(5))/2` is both shorter and has slightly less error.

Reply:
I though about it, but did not want to diverge unnecessarily from the wiki article. But if it is also more accurate, then I am sold.

Reply:
A tiny bit sad about removing the unicode square-root of 5. I get that the unicode symbol actually calls the sqrt function, so one might as well use it directly, but it is also less pretty, and as just noted, it does the same. Why exactly is the ascii version prefered?

Reply:
the unicode sqrt is fine, I was just on a phone